shardingsphere-infra/shardingsphere-infra-optimize/src/test/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/AlterTableStatementFederateRefresherTest.java

```diff
@@ -0,0 +1,234 @@
+        assertTrue(schema.getTables().containsKey("t_order"));
+    }
+    
+    @Test
+    public void renameTableWithRule() throws SQLException {
+        when(materials.getDatabaseType()).thenReturn(new MySQLDatabaseType());
+        renameTableWithRule(new MySQLAlterTableStatement());
+
+        when(materials.getDatabaseType()).thenReturn(new OracleDatabaseType());
+        renameTableWithRule(new OracleAlterTableStatement());
+
+        when(materials.getDatabaseType()).thenReturn(new PostgreSQLDatabaseType());
+        renameTableWithRule(new PostgreSQLAlterTableStatement());
+
+        when(materials.getDatabaseType()).thenReturn(new SQLServerDatabaseType());
+        renameTableWithRule(new SQLServerAlterTableStatement());
+
+        when(materials.getDatabaseType()).thenReturn(new SQL92DatabaseType());
+        renameTableWithRule(new SQL92AlterTableStatement());
+    }
+    
+    private void renameTableWithRule(final AlterTableStatement alterTableStatement) throws SQLException {
+        alterTableStatement.setTable(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order"))));
+        alterTableStatement.setRenameTable(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order_new"))));
+        Map<String, DataSource> dataSourceMap = mock(HashMap.class);
+        TableContainedRule rule = mock(TableContainedRule.class);
+        Collection<ShardingSphereRule> rules = Collections.singletonList(rule);
+        when(materials.getRules()).thenReturn(rules);
+        when(rule.getTables()).thenReturn(Arrays.asList("t_order", "t_order_new"));
+        when(materials.getDataSourceMap()).thenReturn(dataSourceMap);
+        DataSource dataSource = mock(DataSource.class);
+        when(dataSourceMap.get(eq("ds"))).thenReturn(dataSource);
+        Connection connection = mock(Connection.class);
+        when(dataSource.getConnection()).thenReturn(connection);
+        DatabaseMetaData metaData = mock(DatabaseMetaData.class);
+        when(connection.getMetaData()).thenReturn(metaData);
+        ResultSet resultSet = mock(ResultSet.class);
+        when(metaData.getTables(any(), any(), any(), any())).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(false);
+        FederateRefresher<AlterTableStatement> federateRefresher = new AlterTableStatementFederateRefresher();
+        FederateSchemaMetadata schema = buildSchema();
+        federateRefresher.refresh(schema, Collections.singletonList("ds"), alterTableStatement, materials);
+        assertFalse(schema.getTables().containsKey("t_order"));
+        assertTrue(schema.getTables().containsKey("t_order_new"));
+    }
+    
+    @Test
+    public void renameTableWithoutRule() throws SQLException {
+        when(materials.getDatabaseType()).thenReturn(new MySQLDatabaseType());
+        renameTableWithoutRule(new MySQLAlterTableStatement());
+
```

Comment:
Hi, this one is partly duplicated with #10715? Plus, could you look at comments on #10715?